### DISCUSSION TITLE
Why this instead of other microbit BSPs

### DISCUSSION THREAD

**Discussion by winksaville** [lulf/microbit-async#7] [General]
In particular, https://github.com/nrf-rs/microbit. I like what you've done and how it's organized and especially that you're using `probe-rs` instead of the deprecated `probe-run`. But are there other things you'd highlight and what do you see are advantages and disadvantages of "microbit-bsp" vs "microbit"?

Txs in advance!

**Comment by lulf** [comment]
The primary difference is that this crate uses the embassy HAL, whereas the nrf-hal used by the microbit crate is unmaintained. However, the plan is to move the original microbit crate over to embassy hal now, and once done this crate can be archived. Will probably take a while though. See this discussion https://github.com/nrf-rs/nrf-hal/issues/432

**Reply by winksaville** [reply]
Will microbit be moving to probe-rs too?

**Reply by lulf** [reply]
Yes i think it should
